File: 231213/체크판위에서 2/on-the-checkboard-2.js
const fs = require('fs');
const inputAll = () => fs.readFileSync('/dev/stdin').toString().split('\n');
const print = (...arr) => console.log(...arr);

let [mapSize, ...arr] = inputAll()

const [R, C] = mapSize.split(' ').map(elm => Number(elm));
arr = arr.map(line => line.trim().split(' '))

const WHITE = 'W';
const BLACK = 'B';

const reverseTile = (tile) => tile === WHITE ? BLACK : WHITE;

const startTile = arr[0][0];
const endTile = arr[R-1][C-1];

if(startTile === endTile){
    print(0);
    return;
}

let count = 0;

for(let i = 1; i < R - 1; i++){
    for(let j = 1; j < C - 1; j++){
        if(arr[i][j] === reverseTile(startTile)){
            for(let i_2 = i + 1; i_2 < R - 1; i_2++){
                for(let j_2 = j + 1; j_2 < C - 1; j_2++){
                    if(arr[i_2][j_2] === reverseTile(endTile)){
                        count++;
                    }
                }
            }
        }
    }
}

print(count);

File: 231213/체크판위에서 2/on-the-checkboard-2.py
# 왼쪽 상단에서 우측 하단으로 이동
# 시작 - 1 - 2 - 끝을 만족해야함
# 오른쪽 아래로만 이동 가능

WHITE = 'W'
BLACK = 'B'

def reverseTile(tile):
    return WHITE if tile == BLACK else BLACK

N, M = map(int, input().split())
MAP = [input().split() for _ in range(N)]

startTile = MAP[0][0]
endTile = MAP[N-1][M-1]

count = 0

for i in range(1, N - 1):
    for j in range(1, M - 1):
        if MAP[i][j] == reverseTile(startTile):
            for i_2 in range(i + 1, N - 1):
                for j_2 in range(j + 1, M - 1):
                    if MAP[i_2][j_2] == reverseTile(endTile):
                        count += 1

print(count)